Marlin/src/module/tool_change.cpp

```diff
@@ -645,19 +641,14 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
         #endif
 
         #if ENABLED(TOOLCHANGE_FILAMENT_SWAP)
-          if (toolchange_settings.swap_length &&
-            #if ENABLED(PREVENT_COLD_EXTRUSION)
-              (DEBUGGING(DRYRUN) || !thermalManager.targetTooColdToExtrude(active_extruder))
-            #else
-              true
-            #endif
-          ) {
+          if (safeToSwapFil) {
```

Comment:
You forget that there's a loop just above that re-heats the filament and waits, so you can't just use the previous condition here.

Reply:
I was aiming for consistency between the retract and prime moves. If you cant do one, you shouldnt do the other, hence the up front check. In the case of singlenozzle, which does have the heating command, it also has a return statement in the first branch as it cant do anything else, so if the branch to wait for heating is run it wouldnt make it here if this value wasnt true.